### PR TITLE
registry_syncer controller: Implement filters

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-1656?focusedCommentId=15345756&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15345756

/hold

Testing it on my PC ...

/cc @alvaroaleman @stevekuznetsov

### CODE REVIEW

File: pkg/controller/registrysyncer/registrysyncer.go [hunk 1]
```diff
@@ -296,13 +309,46 @@ func (r *reconciler) pullSecret(namespace string) (*corev1.Secret, crcontrolleru
...
+	l *logrus.Entry,
+	excludedImageStreamTags sets.String,
+	excludedImageStreams sets.String,
+	excludedImageStreamNamespaces sets.String,
+	excludedImageStreamNamespacePrefixes sets.String,
+) objectFilter {
+	l = logrus.WithField("subcomponent", "test-input-image-stream-tag-filter")
 	return func(nn types.NamespacedName) bool {
-		// TODO: make the filter for deny-list
+		if excludedImageStreamTags.Has(nn.String()) {
```
Comment (alvaroaleman) [inline pkg/controller/registrysyncer/registrysyncer.go:321]: Can we not push this all down into configuration and instead start thinking about this here already?

For example:
* We certainly do not want to sync anything that comes from a ci-operator namespace
* We do not want to sync anything that is part of openshift itself (is there any way we can determine that)?

IMHO the config is only a fallback for exceptions to the rules we codify here

Comment (hongkailiu) [reply pkg/controller/registrysyncer/registrysyncer.go:321]:
```
oc --context app.ci get ns -o custom-columns=:metadata.name | grep -vE '(origin|kube-system|default)' | grep -Ev '^(openshift|ci-op)'

azure
bastion
bparees
brew-registry-pullsecret
cert-manager
ci
ci-release
ci-release-ppc64le
ci-release-ppc64le-priv
ci-release-priv
ci-release-s390x
ci-release-s390x-priv
ci-rpms
ci-search
ci-stg
cincinnati-ci
clayton-test
clayton-test-1
code-ready
codeready-toolchain
coreos
dedicated-admin
fabric8-services
fcos
hive-integration-access
hongkliu-test
hypershift-toolkit
ibm-gate-secrets
integr8ly
kiegroup
kube-node-lease
kube-public
ocp
ocp-ppc64le
ocp-ppc64le-priv
ocp-priv
ocp-private
ocp-s390x
ocp-s390x-priv
open-cluster-management
osde2e
promecieus
prow-logging
prow-monitoring
prow-monitoring-stage
qe
redhat-integration
rhcos
telemeter
test-credentials
```

```
oc --context api.ci get ns -o custom-columns=:metadata.name | grep -vE '(origin|kube-system|default)' | grep -Ev '^(openshift|ci-op)'

3scaleqe
ablock
acxhvfavbt
adahiya-testing
akaplan
anarayan
anothertest
apavel
app-sre
art
artmetrics
aws-efs-operator-secrets
azure
azure-private
bastion
bastion-ppc64le
bastion-ppc64le-libvirt
bastion-telco
bastion-z
bbguimaraes
bgmlhqgexl
bnemec
bparees
brawilli
brew-registry-pullsecret
bugzilla-operator
bz1784163
cakephp-mysql0
cardil
cblecker
ccx-secrets
cewong-dev
cgwalters
cgwalters-testing
chancez-e2e-35
che-testing
chmouel
ci
ci-byoh
ci-kubernetes-reporting
ci-ln-0lbxl8k
ci-ln-0r262kk
ci-ln-0y9f5sb
ci-ln-1g6r452
ci-ln-2pq9ggk
ci-ln-33bvjht
ci-ln-4y19k4t
ci-ln-4y19k4t-0
ci-ln-4y19k4t-1
ci-ln-4y19k4t-2
ci-ln-6bw7312
ci-ln-70983yk
ci-ln-7dt4j6t
ci-ln-7hckw0k
ci-ln-9spli82
ci-ln-bhkmrcb
ci-ln-cywmtrb
ci-ln-dr88nf2
ci-ln-f4xlkxk
ci-ln-hl7dg32
ci-ln-i94wspk
ci-ln-j9h63ck
ci-ln-n35xvx2
ci-ln-qhr4swk
ci-ln-qjs1nhk
ci-ln-rmd5nc2
ci-ln-rv2sf2k
ci-ln-s9c2xf2
ci-ln-vcng382
ci-ln-ws8rgpk
ci-ln-xgrnp4b
ci-ln-yl6srkt
ci-ln-z979r42
ci-metrics
ci-ocs-operator-secrets
ci-pr-images
ci-release
ci-release-ppc64le
ci-release-ppc64le-priv
ci-release-priv
ci-release-s390x
ci-release-s390x-priv
ci-rpms
ci-stg
ci-test-ewolinet
ci-thanos-operator
cincinnati
cincinnati-ci
cincinnati-ci-public
clayton-test-1
cluster-operator
cluster-secrets-openshift-virtualization
clusterproject0
clusterproject1
clusterproject2
clusterproject3
clusterproject4
clusterproject5
clusterproject6
clusterproject7
clusterproject8
clusterproject9
cnf-qe
cnv-qe
cnv-secrets
code-ready
codeready-toolchain
common-templates-secrets
coreos
coreos-assembler
coreos-devel
crawford-test
crc-bundle-secret
cron
crw-ci-secrets
cvogt
cvp-images
cwyydxfuvu
dancer-mysql0
danwinship
davidvossel
deads-deleteme
deads2k-01
demo2
derekh
dgoodwin-dev
dmartin
dmisharo-dev
dprince-moc
dulek-test
dummy
e2e-oauth-proxy-p8vvh
e2e-oauth-proxy-wkjjz
eb96a609
edge
encryption-gfmtmlcknq
encryption-sbhzr27tgb
fabric8-services
fcos
flaper87-moc
flouwmkabq
fqgnzvfzjx
fuzbvrpodb
hamzy-test-hostaliases
heckinchonker
heimdall-secrets
hello-openshift
hive-integration-access
hive-stage-pullsecret
hongkliu-test
honzaci
http-test2
hypershift-toolkit
ibm-gate-secrets
ibm-registry-keys
image-registry-publishing-bot
integr8ly
integration-qe
interop-qe
ipv6
irosenzw-test
jai-test
jaypoulz
jbieren
jc-crud
jcaianirh-console-tests
jeremy-ssl-junk
jlebon
jlebon-test-1
jwforres-test
k8s-publishing-bot
kam
kiegroup
kind
knative-build
knative-eventing
knative-serving
kni
kube-public
kube-service-catalog
kubevirt-ipi-provider
kuryr-bot
libvirt-ci-ppc64le
lmeyer
lmilbaum-secrets
local-operators
localstorage
loki-ci
management-infra
mansikulkarni96-docker-build
markmc
maszulik
matchbox
metal-testing
metering
metering-prerelease
mkarnik-secrets
mkumatag-bastion-ppc64le-libvirt
mkumatag-test
mrunalp-mirrors
mrunalptest
msimka
muller-bundle-testing
multi-arch-secrets
multiarchocp
mvnmmldqyc
my-mkumatag-secret
my-ripsaw
namespace-bohk
namespace-ywnw
newkargakistest
nmoraitis-prow
nmoraitis-test
noobaa
obulatov
ocp
ocp-future
ocp-ppc64le
ocp-ppc64le-priv
ocp-priv
ocp-private
ocp-s390x
ocp-s390x-priv
ocs
ocs-ci-test
ocsproject0
ocsproject1
ocsproject2
ocsproject3
ocsproject4
ocsproject5
ocsproject6
ocsproject7
ocsproject8
ocsproject9
odo
odo-secrets
olnester
oo-index-images
open-cluster-management
openstack
openstack-vexxhost-secrets
opkwdjkixo
osd-images
osd-operators
osde2e
osde2e-secrets
ovirt
phil1
ppc64le-openstack-installer-43
ppc64le-openstack-installer-44
ppc64le-openstack-installer-45
ppc64le-openstack-installer-46
ppc64le-openstack-secrets
ppc64le-osuosl-ci
ppc64le-osuosl-ci-public
pratik
prj1
prj2
promecieus
psap
ptgnjdlilp
qe
ravid-cnv-provider
redhat-developer
redhat-integration
redhat-rhmi-operator
rhcos
rhcos-devel
rmarasch
robb
rsevilla
rteague-1
save-42
save-43
sd-sre-secrets
search-debug
serenatest
serving-tests
serving-tests-alt
shurley-test-2
squeed
stranger
sttts
sttts-bot
sylgcjnrbe
tekton-pipelines
tekton-triggers-tests
tektoncd-build-916171
telemeter
terminal
test
test-credentials
test-ingress
test-openshift-psap
test-ppc64le-bastion
test1
test2
test6
testaccesspermission
testpvc
tnozicka-acme-controller
tnozicka-acme-operator
tnozicka-dr-e2e-test
tnozicka-images
tnozicka-ocp-code-tools
tnozicka-secrets
tnozicka-test
travier
trown-moc
ttt
tunedover
ucrqsnqrqh
uthapykoej
vdemeest
vikssstest
vrutkovs
winc-test
windows-overview
winmachineman
winoperatordata
wjiang
wking
wxrkghriss
wzheng1
wzheng123
ychnppmbsf
yhui
z1
```

Comment (hongkailiu) [reply pkg/controller/registrysyncer/registrysyncer.go:321]: The one I think we should filter out: `grep -vE '(origin|kube-system|default)' | grep -Ev '^(openshift|ci-op)'`.

@stevekuznetsov 
WDYT?

Comment (hongkailiu) [reply pkg/controller/registrysyncer/registrysyncer.go]: Maybe start with syncing nothing and opt in on the way.
Reduce the possibility of unexpected imports.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**